extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl

```diff
@@ -72,6 +72,19 @@ namespace Microsoft.Windows.DevHome.SDK
         Windows.Foundation.IAsyncOperation<ProviderOperationResult> CloneRepositoryAsync(IRepository repository, String cloneDestination, IDeveloperId developerId);
     };
 
+    [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 2)] interface IRepositoryProvider2
+    requires IRepositoryProvider
+    {
+        String[] GetSearchFieldNames
+        {
+            get;
+        };
+
+        Windows.Foundation.IAsyncOperation<IVector<String> > GetValuesForField(String fieldName, IMapView<String, String> fieldValues, IDeveloperId developerId);
```

Comment:
"Values" here mean repository names or can it be something else?

Reply:
Technically, fieldValues can be anything.  Each extension can handle the input fields however they want.  However, extensions should be able to use every field returned by GetSearchFields.

So, let's say GetSearchFields returns `{Organization, Project}`.  When GetValuesForField is called, the extension should look for, and understand `Organization` and `Project` as potential fields keys and do something with them to return a list of strings.